### ISSUE TITLE
jsx-a11y/no-onchange is deprecated

### ISSUE THREAD

**Issue by stephenlprice** [gatsbyjs/gatsby#31895]
## Description

I haven't seen this mentioned in other issues or discussions but the warning for this rule `jsx-a11y/no-onchange` should be deprecated as indicated by the docs for [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md).

### Steps to reproduce

The following would trigger the warning:

```jsx
<select value={subject} onChange={handleChange}>
  <option>Apples</option>
  <option>Oranges</option>
  <option>Peaches</option>
  <option>Bananas</option>
</select>
```

The following passes but setting state does not seem to work with onBlur, at least it seems to give [problems to others as well](https://github.com/sveltejs/svelte/issues/4946). Maybe the handleBlur function needs to differ from the handleChange function in some way but I am not sure how.

```jsx
<select value={subject} onBlur={handleBlur}>
  <option>Apples</option>
  <option>Oranges</option>
  <option>Peaches</option>
  <option>Bananas</option>
</select>
```
```jsx
<select value={subject} onChange={handleChange} onBlur={handleBlur}>
  <option>Apples</option>
  <option>Oranges</option>
  <option>Peaches</option>
  <option>Bananas</option>
</select>
```

### Expected result

The rule is deprecated as indicated by the [docs](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md).

### Actual result

The first select tag triggered a warning. I could use `// eslint-disable-next-line` but I rather avoid it when possible.

### Environment
 `gatsby@3.2.1` uses `eslint-plugin-jsx-a11y@6.4.1`

  System:
    OS: macOS 11.4
    CPU: (8) arm64 Apple M1
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.9.0 - ~/.nvm/versions/node/v15.9.0/bin/node
    npm: 7.5.3 - ~/.nvm/versions/node/v15.9.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Safari: 14.1.1
  npmPackages:
    gatsby: ^3.2.1 => 3.2.1 
    gatsby-plugin-mdx: ^2.3.0 => 2.3.0 
    gatsby-plugin-page-creator: ^3.3.0 => 3.3.0 
    gatsby-plugin-react-helmet: ^4.3.0 => 4.3.0 
    gatsby-plugin-sass: ^4.2.0 => 4.2.0 
    gatsby-plugin-sharp: ^3.4.0 => 3.4.0 
    gatsby-plugin-svgr: ^3.0.0-beta.0 => 3.0.0-beta.0 
    gatsby-remark-images: ^5.1.0 => 5.1.0 
    gatsby-remark-prismjs: ^5.0.0 => 5.0.0 
    gatsby-source-filesystem: ^3.2.0 => 3.2.0 
  npmGlobalPackages:
    gatsby-cli: 3.4.1
